### PR TITLE
KB-4212 | Avoid Auto batch creation for Invite-Only Assessment from Post Publish Flink Job

### PR DESCRIPTION
**Pull request by tarentomaheshvakkund**

### CODE REVIEW

File: post-publish-processor/src/main/scala/org/sunbird/job/postpublish/helpers/BatchCreation.scala [hunk 1]
```diff
@@ -69,7 +69,14 @@ trait BatchCreation {
       val trackableObj = JSONUtil.deserialize[java.util.Map[String, AnyRef]](trackableStr)
       val trackingEnabled = trackableObj.getOrDefault("enabled", "No").asInstanceOf[String]
       val autoBatchCreateEnabled = trackableObj.getOrDefault("autoBatch", "No").asInstanceOf[String]
-      val trackable = (StringUtils.equalsIgnoreCase(trackingEnabled, "Yes") && StringUtils.equalsIgnoreCase(autoBatchCreateEnabled, "Yes"))
+      var trackable = (StringUtils.equalsIgnoreCase(trackingEnabled, "Yes") && StringUtils.equalsIgnoreCase(autoBatchCreateEnabled, "Yes"))
+      val courseCategory = metadata.getOrDefault("courseCategory", "").asInstanceOf[String]
+      if (trackable) {
+        if (StringUtils.containsIgnoreCase(courseCategory, "Invite-Only")) {
+          logger.info("CourseCategory for " + identifier + " : " + courseCategory)
```
Comment (karthik-tarento) [inline post-publish-processor/src/main/scala/org/sunbird/job/postpublish/helpers/BatchCreation.scala]: Update the logger -- do mention that it is setting trackable as false

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: post-publish-processor/src/main/scala/org/sunbird/job/postpublish/helpers/BatchCreation.scala (hunk 1 above)
Comment (tarentomaheshvakkund) [reply post-publish-processor/src/main/scala/org/sunbird/job/postpublish/helpers/BatchCreation.scala]: Have updated the log.

Comment (karthik-tarento) [reply post-publish-processor/src/main/scala/org/sunbird/job/postpublish/helpers/BatchCreation.scala]: one logger - 
logger.info("CourseCategory for " + identifier + " : " + courseCategory + ", setting trackable to false");

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: post-publish-processor/src/main/scala/org/sunbird/job/postpublish/helpers/BatchCreation.scala (hunk 1 above)
Comment (tarentomaheshvakkund) [reply post-publish-processor/src/main/scala/org/sunbird/job/postpublish/helpers/BatchCreation.scala]: changed.

**Review by karthik-tarento — APPROVED** [review]

**Pull request merged**